### ISSUE TITLE
Migration of previous cache path doesn't work

### ISSUE THREAD

**Issue by cipherCOM** [SDWebImage/SDWebImage#2993]
### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 13.4
 SDWebImage Version      | from 3.7.1 to 5.7.3
 Integration Method      | manually
 Xcode Version           | Xcode 11.4
 Repro rate              | all the time (100%)
 Repro with our demo prj | presumably, but didn't check
 Demo project link       | not needed

### Issue Description and Steps

If you run e.g. version 3.7.1 and then update to 5.7.3 the old image cache folder won't get moved as described in the readme.

### Proposed bugfix

```--- a/code/Pods/SDWebImage/SDWebImage/Core/SDImageCache.m
+++ b/code/Pods/SDWebImage/SDWebImage/Core/SDImageCache.m
@@ -132,8 +132,8 @@
         dispatch_once(&onceToken, ^{
             // ~/Library/Caches/com.hackemist.SDImageCache/default/
             NSString *newDefaultPath = [[[self userCacheDirectory] stringByAppendingPathComponent:@"com.hackemist.SDImageCache"] stringByAppendingPathComponent:@"default"];
-            // ~/Library/Caches/default/com.hackemist.SDWebImageCache.default/
-            NSString *oldDefaultPath = [[[self userCacheDirectory] stringByAppendingPathComponent:@"default"] stringByAppendingPathComponent:@"com.hackemist.SDWebImageCache.default"];
+            // ~/Library/Caches/com.hackemist.SDWebImageCache.default/
+            NSString *oldDefaultPath = [[self userCacheDirectory] stringByAppendingPathComponent:@"com.hackemist.SDWebImageCache.default"];
             dispatch_async(self.ioQueue, ^{
                 [((SDDiskCache *)self.diskCache) moveCacheDirectoryFromPath:oldDefaultPath toPath:newDefaultPath];
         });

**Comment by dreampiggy** [comment]
Your path is 3.x path.

In 4.x the cache path is actually:

```
 ~/Library/Caches/default/com.hackemist.SDWebImageCache.default/
```

It contains a `default`. You can check again. 

The migration code is temp for 5.x, and will remove in 6.x, we do not migrate any 3.x version.

**Comment by cipherCOM** [comment]
I thought I checked it (looked into `4.4.8`) but you're right. Sad only one major jump is supported. Would an additional 'check if file exist' for older cache paths not be a good idea? One wouldn't have to move the directory twice but rather find the correct old path. I think this is favourable instead of what happens now, that users have a dangling 3.x cache path with multiple MBs on their disk if the developer did not notice it.
Apart from that, thanks for the hint with `moveCacheDirectoryFromPath:toPath:` and glad to see `SDWebImage` got quite the features (we emulated multiple image sizes with 3.x which is now directly supported). Fantastic work!
